tests/test_metadata_utils.py

```diff
@@ -41,11 +49,88 @@ def test_resolve_variant_warehouse_db_name_from_database(self):
+        ensure_taxonomy_is_in_evapro(db_handle, 9606)
+        # Checks that the taxonomy does not exist then
+        db_handle.cursor().execute.assert_any_call('SELECT taxonomy_id FROM evapro.taxonomy WHERE taxonomy_id=9606')
+        # Insert the taxonomy with the appropriate species name
+        db_handle.cursor().execute.assert_any_call(
+            'INSERT INTO evapro.taxonomy(taxonomy_id, common_name, scientific_name, taxonomy_code, eva_name) '
+            'VALUES (%s, %s, %s, %s, %s)',
+            (9606, 'human', 'Homo sapiens', 'hsapiens', 'human')
+        )
+        db_handle.reset_mock()
+        # Can override the eva_name
+        ensure_taxonomy_is_in_evapro(db_handle, 9606, 'tortoise')
+        db_handle.cursor().execute.assert_any_call(
+            'INSERT INTO evapro.taxonomy(taxonomy_id, common_name, scientific_name, taxonomy_code, eva_name) '
+            'VALUES (%s, %s, %s, %s, %s)',
+            (9606, 'human', 'Homo sapiens', 'hsapiens', 'tortoise')
+        )
+
+    def test_insert_assembly_in_evapro_no_insert(self):
+        db_handle = MagicMock()
+        db_handle.cursor().__enter__().fetchall.return_value = (9606,)
+        with patch('ebi_eva_common_pyutils.metadata_utils.insert_taxonomy') as mock_insert:
+            ensure_taxonomy_is_in_evapro(db_handle, 9606)
+        mock_insert.assert_not_called()
+
+    def test_insert_assembly_in_evapro(self):
+        db_handle = MagicMock()
+        with patch('ebi_eva_common_pyutils.metadata_utils.get_all_results_for_query') as mock_results:
+            # This will be used when retrieving the assembly set id after it's been inserted
+            assembly_set_id = 27
+            mock_results.return_value = ((assembly_set_id,),)
+            insert_assembly_in_evapro(db_handle, 9606, 'GCA_000001405.15', 'GRCh38', 'grch38')
+            db_handle.cursor().execute.assert_any_call(
+                'INSERT INTO evapro.assembly_set(taxonomy_id, assembly_name, assembly_code) VALUES (%s, %s, %s)',
+                (9606, 'GRCh38', 'grch38')
+            )
+            db_handle.cursor().execute.assert_any_call(
+                'INSERT INTO evapro.accessioned_assembly(assembly_set_id, assembly_accession, assembly_chain, assembly_version) VALUES (%s,%s,%s,%s)',
+                (assembly_set_id, 'GCA_000001405.15', 'GCA_000001405', '15')
+            )
+
+    def test_update_accessioning_status(self):
+        db_handle = MagicMock()
+        update_accessioning_status(db_handle, 'GCA_000001405.15', in_accessioning_flag=True)
+        db_handle.cursor().execute.assert_any_call(
+            ("INSERT INTO evapro.assembly_accessioning_store_status "
+             "SELECT * FROM "
+             "(SELECT cast('GCA_000001405.15' as text) as assembly_accession, cast('True' as boolean) as loaded) temp "
+             "WHERE assembly_accession NOT IN "
+             "(SELECT assembly_accession FROM evapro.assembly_accessioning_store_status)")
+        )
```

Comment:
Tbh methods like this that really only execute SQL, I don't think it's worth testing... I'd rather just test the Python logic even if it's minimal rather than tests that will break if we just change the whitespace or something - at least until we start using an actual postgresql test framework, so we can actually test data changes (maybe we should make a ticket to start doing this, it will become increasingly important as we move metadata functionality out of perl).

If you want to keep the tests, one way we could maybe make them less brittle is to assert the calls happen and that relevant substrings exist in the arguments, e.g. ensure we insert into the right tables or something.

Reply:
My view would be that SQL is code and we should test it as well. But if we want to test the functionality of it rather than the syntax, the best way is to have database to connect to and check the operation occurred.